json_pointer.c

```diff
@@ -0,0 +1,242 @@
+/*
+ * $Id: json_pointer.c,v 1.20 2016/10/25 03:24:50 mclark Exp $
+ *
+ * Copyright (c) 2016 Metaparadigm Pte. Ltd.
+ * Michael Clark <michael@metaparadigm.com>
+ *
+ * This library is free software; you can redistribute it and/or modify
+ * it under the terms of the MIT license. See COPYING for details.
+ *
+ *
+ * Copyright (c) 2008-2009 Yahoo! Inc.  All rights reserved.
+ * The copyrights to the contents of this file are licensed under the MIT License
+ * (http://www.opensource.org/licenses/mit-license.php)
+ */
+
+#include "config.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <string.h>
+#include <ctype.h>
+
+#include "json_pointer.h"
+
+/**
+ * JavaScript Object Notation (JSON) Pointer
+ *   RFC 6901 - https://tools.ietf.org/html/rfc6901
+ */
+
+static void string_replace_all_occurrences_with_char(char *s, const char *occur, char repl_char)
+{
+	int slen = strlen(s);
+	int skip = strlen(occur) - 1; /* length of the occurence, minus the char we're replacing */
+	char *p = s;
+	while ((p = strstr(p, occur))) {
+		*p = repl_char;
+		p++;
+		slen -= skip;
+		memcpy(p, (p + skip), slen - (p - s) + 1); /* includes null char too */
+	}
+}
+
+static int is_valid_index(struct json_object *jo, const char *path, int32_t *idx)
+{
+	int i, len = strlen(path);
+	/* this code-path optimizes a bit, for when we reference the 0-9 index range in a JSON array
+	   and because leading zeros not allowed */
+	if (len == 1) {
+		int ret = isdigit(path[0]);
+		ret ? (*idx = (path[0] - '0')) : (errno = EINVAL);
```

Comment:
damn ; you're right ; thanks for catching this ; 
i think i added this after a quick refactor during writing this ; `is_valid_index()` initially was meant to handle if `path` is a valid number conforming to RFC ; but then i added the json_object param